src/components/BotAppearance/BotAppearance.tsx

```diff
@@ -70,13 +84,22 @@ export const BotAppearancePage = () => {
   useEffect(() => {
     if (sources && !isInitialData.current) {
       isInitialData.current = true
+
+      if (sources.accentColour == null) {
+        sources.accentColour = '#ffffff'
+      }
```

Comment:
Avoid mutating objects like this; instead, use ﻿`||` for fallback value and `﻿===` instead of ﻿`==`.

Also, we should migrate the data in the database rather than setting it in the client; if not possible, leave it as an empty value.